crates/history/src/utils.rs

```diff
@@ -4,12 +4,30 @@ use std::sync::atomic::{AtomicU32, Ordering};
 
 use wasm_bindgen::throw_str;
 
+#[cfg(not(target_arch = "wasm32"))]
 pub(crate) fn get_id() -> u32 {
     static ID_CTR: AtomicU32 = AtomicU32::new(0);
 
     ID_CTR.fetch_add(1, Ordering::SeqCst)
 }
 
+#[cfg(target_arch = "wasm32")]
+pub(crate) fn get_id() -> u32 {
+    static ID_CTR: AtomicU32 = AtomicU32::new(0);
+    static INIT: std::sync::Once = std::sync::Once::new();
+
+    INIT.call_once(|| {
+        let mut start: [u8; 4] = [0; 4];
+        // If it fails then the start is not or only partly filled.
+        // But since this method should not fail, we take what we get.
+        let _ = getrandom::getrandom(&mut start);
+        // Using a high initial value is not an issue as `fetch_add` does wrap around.
+        ID_CTR.store(u32::from_ne_bytes(start), Ordering::SeqCst);
+    });
+
+    ID_CTR.fetch_add(1, Ordering::SeqCst)
```

Comment:
```suggestion
    let mut start: [u8; 4] = [0; 4];
    let _ = getrandom::getrandom(&mut start);
    u32::from_ne_bytes(start)
```

I think we can omit the initialisation logic and get a random value as id every time.
This might help us to save bundle size and I feel this might be safer.

However, the 2 methods should have the same probability for the scenarios I can come up with.
So I am fine with the current method as well.

Reply:
I don't think it's a good idea to use a random value each time because then there could be a collision, which is currently impossible (unless 2^32 id's are generated).

So I'll leave it at the current method.

Reply:
> I don't think it's a good idea to use a random value each time because then there could be a collision, which is currently impossible (unless 2^32 id's are generated).

`2^32` is only the best case scenario, where it only happens if the start id is assigned the same as the last id.

Suppose `64` is selected for a start id with 10 navigation, then ids `64~73` are given out.

For another page to collide with currently assigned id within 10 navigations, the start id needs to be within `55~73`.

Suppose the id is `u8`, which will make the calculation easier.
Assume the start id is selected based on `1/256` for every possibility, then the probability is `(73 - 55 + 1 (both ends are inclusive) + 1 (assigning 11th id for the second navigation)) / 256`, which is `20 / 256`.

If every id is random, then for each unique id given out, the likelihood of collision increases by `1/256`. After 20 ids are assigned, the 21st id is `20/256`.

So the probability is the same.